source/common/buffer/buffer_impl.h

```diff
@@ -10,6 +10,43 @@
 namespace Envoy {
 namespace Buffer {
 
+/**
+ * An implementation of BufferFragment where a releasor callback is called when the data is
+ * no longer needed.
+ */
+class BufferFragmentImpl : public BufferFragment {
+public:
+  /**
+   * Creates a new wrapper around the externally owned <data> of size <size>.
+   * The caller must ensure <data> is valid until releasor() is called, or for the lifetime of the
+   * fragment. releasor() is called with <data>, <size> and <this> to allow caller to delete
+   * the fragment object.
+   * @param data external data to reference
+   * @param size size of data
+   * @param releasor a callback function to be called when data is no longer needed.
+   */
+  BufferFragmentImpl(const void* data, size_t size,
+                     std::function<void(const void*, size_t, const BufferFragmentImpl*)> releasor)
+      : data_(data), size_(size), releasor_(releasor) {}
+
+  const void* data() const override { return data_; }
```

Comment:
nit: Envoy code style is to put overridden methods under a comment that looks like:
```
// Buffer::BufferFragment
```

Then move other methods in the same permission level above the comment so it's clear from a quick glance they are not overriden.

Reply:
Done.